### ISSUE TITLE
Proposal: Add tests

### ISSUE THREAD

**Issue by kbariotis** [serverless/examples#158]
I would suggest adding tests to each one of the examples. Testing is a big subject and lots of people that are interested in a serverless architecture may be wondering how one is being tested. Your recommendation on the subject will be a big deal, imho.

Thank you

**Comment by darrenhgc** [comment]
I agree that it'd be nice to see some examples that include tests. That being said, everyone has a different approach to testing. Both in terms of libraries used, if any, and methodologies. While I feel that examples with tests would be helpful, the task should be approached with care and on the coat tails of a bit of discussion.

**Comment by kbariotis** [comment]
@darrenhgc absolutely. Serverless has already an official [testing guide](https://serverless.com/framework/docs/providers/aws/guide/testing/) that could be a good starting point. However, like you said, there are many solutions to the same problem so proper discussion is required.

**Comment by darrenhgc** [comment]
I'm game. I think [aws-node-twillio-send-text-message](https://github.com/serverless/examples/tree/master/aws-node-twilio-send-text-message) would be a good subject to base our conversation around. If there is another example in mind let me know.  I'd be willing to introduce some tests to the project after a bit of discussion. I'd be seeking input about directory structure and tooling. For reference about tooling, the serverless project is using `chai`, `mocha`, `sinon` and `proxyquire`.

**Comment by darrenhgc** [comment]
Feedback encouraged for #160. Thanks in advance

**Comment by leeeomaaax** [comment]
@darrenhgc, nice work. Things I would also do:
- write an integration test using lambda-wrapper or docker-lambda
- remove those dev dependencies from the package (eg, sinon, chai)
```
package:
  exclude:
    - node_modules/**
    - '!node_modules/twilio/**'
```

**Comment by leeeomaaax** [comment]
@darrenhgc , now I'm trying to write tests using proxyquire to stub a s3.getObject, following the example [here](https://gist.github.com/ifraixedes/3330ce0edf9286234b04).

**Comment by Saphyel** [comment]
I think this is the most important feature they can provide...
